nifi-nar-bundles/nifi-update-attribute-bundle/nifi-update-attribute-processor/src/main/java/org/apache/nifi/processors/attributes/UpdateAttribute.java

```diff
@@ -489,17 +546,32 @@ private FlowFile executeActions(final ProcessSession session, final ProcessConte
         final Map<String, String> attributesToUpdate = new HashMap<>(actions.size());
         final Set<String> attributesToDelete = new HashSet<>(actions.size());
 
+        final Map<String, String> statefulAttributesToSet;
+
+        if (statefulAttributes != null){
+            statefulAttributesToSet = new HashMap<>();
+        } else {
+            statefulAttributesToSet = null;
+        }
+
+
         // go through each action
         for (final Action action : actions.values()) {
             if (!action.getAttribute().equals(DELETE_ATTRIBUTES.getName())) {
```

Comment:
I didn't write this piece of code so I'm not entirely sure but I highly doubt it since its been on master for a while and hasn't been a problem so far.